### PR TITLE
New random number generator

### PR DESCRIPTION
**Pull request by jbmouret**
With C++11 and thread_local

A different random generator is created for int / float / Gaussian. Not sure it's perfect but it should be OK for our purpose.

### CODE REVIEW

File: sferes/ea/eps_moea.hpp [hunk 1]
```diff
@@ -63,8 +63,9 @@ namespace sferes {
         this->_pop.resize(Params::pop::size);
         parallel::p_for(parallel::range_t(0, this->_pop.size()),
                         random<Phen>(this->_pop));
+        std::cout<<"random done"<<std::endl;
```
Comment (costashatz) [inline sferes/ea/eps_moea.hpp]: I guess this should be removed, right?

File: sferes/ea/eps_moea.hpp [hunk 2]
```diff
@@ -63,8 +63,9 @@ namespace sferes {
...
         this->_eval_pop(this->_pop, 0, this->_pop.size());
-
+        std::cout<<"initial eval ok"<<std::endl;
```
Comment (costashatz) [inline sferes/ea/eps_moea.hpp]: This one also?

File: sferes/gen/evo_float.hpp [hunk 3]
```diff
@@ -165,7 +165,8 @@ namespace sferes {
       template<typename Ev>
       struct CrossOver_f<Ev, recombination> {
         void operator()(const Ev& f1, const Ev& f2, Ev &c1, Ev &c2) {
-          size_t k = misc::rand<unsigned int>(f1.size());
+          size_t k = misc::rand<size_t>(f1.size());
+	  assert(k < f1.size());
```
Comment (costashatz) [inline sferes/gen/evo_float.hpp]: The indentation seems wrong. Maybe the usual tab/space issue?

**Comment by jbmouret** [comment]
Should we merge?

**Comment by costashatz** [comment]
> Should we merge?

I think we discuss a bit more this commit: 04f2873. If we want to drop C++98 support and require C++11 or bigger (which I am fine with), then we need to completely remove the `--cpp11` option from wscript. If not, we should provide a fallback random generator (most probably the old one) when only C++98 is available.. What do you suggest?

**Comment by jbmouret** [comment]
Merge now?

**Comment by VladislavTempez** [comment]
eps moea is in no parallel. (don't know if it's intended)

**Comment by costashatz** [comment]
**Seems OK to me**

**Pull request merged**